### PR TITLE
Sync schedules

### PR DESCRIPTION
**Pull request by benwiley**
*** Rock, core, and give all have pull requests

1.) Make the defined value type guids available on the client
2.) Sync payment details
3.) Sync defined values first since they are needed for other entity syncing (priority sync)
4.) Many updates to allow operation with Rock's new paymentDetail schema change
5.) Change to using the correct field of a personAlias, which is simply the Id field
6.) Sync saved accounts and groups
7.) CampusId of person is updated through family group

### CODE REVIEW

File: _source/server/collections/person.coffee [hunk 1]
```diff
@@ -62,6 +54,11 @@ Rock.person.create = (person) ->
       return
 
     newId = Number result.content
+    Apollos.people.update _id: mongoId,
```
Comment (jbaxleyiii) [inline _source/server/collections/person.coffee]: Can we async this?

File: _source/server/collections/savedAccount.coffee [hunk 2]
```diff
@@ -1,5 +1,123 @@
...
+      "This savedAccount contains an Id, which means Rock already has it"
+
+  if not (person.FirstName and person.LastName)
+    Apollos.debug "Cannot create person on Rock without first and last name"
+    return
+
+  delete person.Id
+  person.Guid or= Rock.utilities.makeNewGuid()
+
+  Apollos.people.update
```
Comment (jbaxleyiii) [inline _source/server/collections/savedAccount.coffee]: Can we async this to make it non blocking?

File: _source/server/collections/savedAccount.coffee [hunk 3]
```diff
@@ -1,5 +1,123 @@
...
+
+  Rock.apiRequest "POST", "api/People", person, (error, result) ->
+    if error
+      Apollos.debug "Rock person create failed:"
+      Apollos.debug "THE ERROR:", error
+      Apollos.debug "THE DATA SENT:", person
+      return
+
+    newId = Number result.content
+    Apollos.people.update _id: mongoId,
```
Comment (jbaxleyiii) [inline _source/server/collections/savedAccount.coffee]: Can we async this to make it non blocking?

File: _source/server/collections/savedAccount.coffee [hunk 4]
```diff
@@ -1,5 +1,123 @@
...
+      $set:
+        personId: newId
+        updatedBy: Rock.name
+
+    user = Apollos.users.findOne personGuid: new RegExp(person.Guid, "i")
+
+    if user
+      Apollos.users.update user._id,
```
Comment (jbaxleyiii) [inline _source/server/collections/savedAccount.coffee]: Can we async this to make it non blocking?

File: _source/server/collections/user.coffee [hunk 5]
```diff
@@ -166,7 +166,10 @@ Rock.user.create = (user) ->
       Apollos.debug error
       return
 
-    Apollos.user.update result.data, Rock.name
+    Apollos.users.update _id: mongoId,
```
Comment (jbaxleyiii) [inline _source/server/collections/user.coffee]: Can we async this to make it non blocking?

File: _source/server/services/give.coffee [hunk 6]
```diff
@@ -1,7 +1,35 @@
 
-if not Apollos.give
+if not Apollos.give or not Apollos.schedule
```
Comment (jbaxleyiii) [inline _source/server/services/give.coffee]: Another reason I think schedule should be under give? It is part of the give package to schedule gifts

File: _source/server/collections/person.coffee (hunk 1 above)
Comment (benwiley) [inline _source/server/collections/person.coffee]: Done

File: _source/server/collections/savedAccount.coffee (hunk 2 above)
Comment (benwiley) [inline _source/server/collections/savedAccount.coffee]: Done

File: _source/server/collections/savedAccount.coffee (hunk 3 above)
Comment (benwiley) [inline _source/server/collections/savedAccount.coffee]: Done

File: _source/server/collections/savedAccount.coffee (hunk 4 above)
Comment (benwiley) [inline _source/server/collections/savedAccount.coffee]: Done

File: _source/server/collections/user.coffee (hunk 5 above)
Comment (benwiley) [inline _source/server/collections/user.coffee]: Done

**Comment by benwiley** [comment]
PRing into alpha instead of master

**Pull request closed without merging**